gui_scripts/gui_helpers/general_helpers.py

```diff
@@ -202,3 +202,178 @@ def stop(self):
             self.paused = False
             self.output_hints_signal.emit('Stopping simulation from thread')
         self.pause_condition.wakeOne()
+
+
+class DirectoryTreeView(QtWidgets.QTreeView):
+    """
+    Sets up a new directory tree view
+    """
+    item_double_clicked_sig = QtCore.pyqtSignal(QtCore.QModelIndex)
+
+    def __init__(self, file_model, parent=None):
+        super().__init__(parent)
+        self.setSelectionBehavior(QtWidgets.QTreeView.SelectRows)
+        self.setSelectionMode(QtWidgets.QTreeView.SingleSelection)
+
+        self.model = file_model
+        self.setModel(self.model)
+
+        self.copied_path = None
+        self.is_directory = False
+        self.is_cut_operation = False
+
+    def copy_item(self, source_index, cut=False):
+        """
+        Implements the copy/cut operations in the directory tree view
+
+        :param source_index:    the QModelIndex of the item to be copied or cut
+        :param cut:             boolean to choose between copying and cutting
+        """
+        # Store the source path and determine if it's a directory
+        self.copied_path = self.model.filePath(source_index)
```

Comment:
I missed this until after I submitted the review, but typically we also list parameters exclusively. So, whatever the parameter is called in the filePath method is should be listed. For example: function_param=source_index

Reply:
Relaxing it for an external library in this case seems fine. In general, we have found this has helped avoid running into bugs in the future when things scale